internal/graph/cached_resolver_test.go

```diff
@@ -0,0 +1,513 @@
+				ContextualTuples: []*openfgav1.TupleKey{
+					{
+						Object:   "document:aaa",
+						Relation: "reader",
+						User:     "user:XYZ",
+					},
+					{
+						Object:   "document:xxx",
+						Relation: "reader",
+						User:     "user:XYZ",
+					},
+				},
+			},
+			subsequentReq: &ResolveCheckRequest{
+				StoreID:              "12",
+				AuthorizationModelID: "33",
+				TupleKey:             tuple.NewTupleKey("document:abc", "reader", "user:XYZ"),
+				ContextualTuples: []*openfgav1.TupleKey{
+					{
+						Object:   "document:aaa",
+						Relation: "reader",
+						User:     "user:XYZ",
+					},
+					{
+						Object:   "document:xxx",
+						Relation: "reader",
+						User:     "user:XYZ",
+					},
+					{
+						Object:   "document:yyy",
+						Relation: "reader",
+						User:     "user:XYZ",
+					},
+				},
+			},
+			setInitialResult: func(mock *MockCheckResolver, request *ResolveCheckRequest) {
+				mock.EXPECT().ResolveCheck(ctx, request).Times(1).Return(result, nil)
+			},
+			setTestExpectations: func(mock *MockCheckResolver, request *ResolveCheckRequest) {
+				mock.EXPECT().ResolveCheck(ctx, request).Times(1).Return(result, nil)
+			},
+		},
+		{
+			name: "first_contextual_tuples_then_no_contextual_tuples_does_not_return_results_from_cache",
+			initialReq: &ResolveCheckRequest{
+				StoreID:              "12",
+				AuthorizationModelID: "33",
+				TupleKey:             tuple.NewTupleKey("document:abc", "reader", "user:XYZ"),
+				ContextualTuples: []*openfgav1.TupleKey{
+					{
+						Object:   "document:aaa",
+						Relation: "reader",
+						User:     "user:XYZ",
+					},
+					{
+						Object:   "document:xxx",
+						Relation: "reader",
+						User:     "user:XYZ",
+					},
+				},
+			},
+			subsequentReq: &ResolveCheckRequest{
+				StoreID:              "12",
+				AuthorizationModelID: "33",
+				TupleKey:             tuple.NewTupleKey("document:abc", "reader", "user:XYZ"),
+				ContextualTuples:     []*openfgav1.TupleKey{},
+			},
+			setInitialResult: func(mock *MockCheckResolver, request *ResolveCheckRequest) {
+				mock.EXPECT().ResolveCheck(ctx, request).Times(1).Return(result, nil)
+			},
+			setTestExpectations: func(mock *MockCheckResolver, request *ResolveCheckRequest) {
+				mock.EXPECT().ResolveCheck(ctx, request).Times(1).Return(result, nil)
+			},
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockResolver := NewMockCheckResolver(ctrl)
+			initialReq := req
+			if test.initialReq != nil {
+				initialReq = test.initialReq
+			}
+			test.setInitialResult(mockResolver, initialReq)
+
+			// expect first call to result in actual resolve call
+			dut := NewCachedCheckResolver(mockResolver,
+				WithLogger(logger.NewNoopLogger()),
+				WithMaxCacheSize(10))
+			defer dut.Close()
+
+			_, _ = dut.ResolveCheck(ctx, initialReq)
+
+			newCtrl := gomock.NewController(t)
```

Comment:
Not needed or used. The `ctrl` above is used for the `mockResolver`.

Reply:
removed